packages/repo/src/types.ts

```diff
@@ -3,32 +3,48 @@ import { schema as common, def as commonDef } from '@atproto/common'
 import { CID } from 'multiformats'
 import BlockMap from './block-map'
 import { RepoRecord } from '@atproto/lexicon'
+import CidSet from './cid-set'
 
 // Repo nodes
 // ---------------
 
 const unsignedCommit = z.object({
   did: z.string(),
-  version: z.number(),
-  prev: common.cid.nullable(),
+  version: z.literal(3),
```

Comment:
ah, we are doing the full (minor) breaking repo version change in this PR.

it is kind of complex and unexpected to me that we will have a time period with both v2 and v3 repos floating around. I guess not that big of a deal b/c they are pseudo-compatible.

Reply:
That's basically my thought^^

The alternative is to take on a decent amount of downtime to migrate all of the repos :/